### PR TITLE
Handle two clients on server, receive and send package to clients

### PR DESCRIPTION
**Pull request by ghost**
server handle two clients, receive packet and send to client
update IT test for 2 clients, check echo
add AppClientHandler - class with main client loop (send and receive packet)

### CODE REVIEW

File: client/src/main/java/com/blackship/battlesheep/communication/network/AppClientHandler.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.blackship.battlesheep.communication.network;
+
+import com.blackship.battlesheep.communication.network.packet.NetworkPacketMove;
+import com.blackship.battlesheep.communication.network.packet.PacketFactory;
+
+import java.io.IOException;
+import java.time.LocalTime;
+
+public class AppClientHandler {
```
Comment (mhbratek) [inline client/src/main/java/com/blackship/battlesheep/communication/network/AppClientHandler.java:9]: JavaDoc.

File: client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java [hunk 2]
```diff
@@ -18,39 +18,30 @@
  */
 public class AppClientSocketIT {
 
-    private static Boolean receivedPacketIsTheSameAsSent;
-
-    @Test
+    @Test(timeOut = 2000)
```
Comment (mhbratek) [inline client/src/test/java/com/blackship/battlesheep/communication/network/AppClientSocketIT.java:21]: :+1:

**Review by mhbratek — APPROVED** [review]

File: client/src/main/java/com/blackship/battlesheep/communication/network/AppClientCommunicationHandler.java [hunk 3]
```diff
@@ -36,14 +37,15 @@ void connect() throws IOException {
...
     }
 
     void write(Packet packet) throws IOException {
-        clientWriter.write(new NetworkPacketConverter().toByte(packet));
+        clientWriter.write(clientPacketConverter.toByte(packet));
     }
 
     Packet read() throws IOException, ClassNotFoundException {
-        return new NetworkPacketConverter().toPacket(clientReader.read());
+        return clientPacketConverter.toPacket(clientReader.read());
```
Comment (arturAR) [inline client/src/main/java/com/blackship/battlesheep/communication/network/AppClientCommunicationHandler.java:48]: 👍

File: client/src/main/java/com/blackship/battlesheep/communication/network/AppClientHandler.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    private AppClientCommunicationHandler clientCommunicationHandler;
+
+    AppClientHandler(AppClientCommunicationHandler clientCommunicationHandler) {
+        this.clientCommunicationHandler = clientCommunicationHandler;
+    }
+
+    public void startClientLoop() throws IOException, ClassNotFoundException {
```
Comment (arturAR) [inline client/src/main/java/com/blackship/battlesheep/communication/network/AppClientHandler.java:17]: Loop?

File: server/src/test/java/com/blackship/blacksheep/communication/network/AppServerSocketIT.java [hunk 5]
```diff
@@ -20,9 +20,34 @@
...
+            byte[] receivedPacketAsBytes = new NetworkReader(givenClient.getInputStream()).read();
+            Packet receivedPacket = new NetworkPacketConverter().toPacket(receivedPacketAsBytes);
+
+            samePacketAsSent = EqualsBuilder.reflectionEquals(givenPacketToSend, receivedPacket);
+        } catch (IOException | InterruptedException | ClassNotFoundException e) {
+            e.printStackTrace();
+            samePacketAsSent = false;
+        }
+
+        return samePacketAsSent;
```
Comment (arturAR) [inline server/src/test/java/com/blackship/blacksheep/communication/network/AppServerSocketIT.java:47]: Why not void?

**Review by arturAR — APPROVED** [review]

**Pull request merged**